context-propagation/src/main/java/io/micrometer/context/DefaultContextSnapshot.java

```diff
@@ -90,6 +90,10 @@ private static <V> Map<Object, Object> setThreadLocal(Object key, V value, Threa
         return previousValues;
     }
 
+    static Scope setAllThreadLocalsFrom(Object context, ContextRegistry registry) {
+        return ContextSnapshot.captureAll(registry, context).setThreadLocals();
+    }
```

Comment:
But this doesn't really bypass the creation of a ContextSnapshot, as the Javadoc says and as it can be. Essentially it should be the same underlying logic as `setThreadLocalsFrom` but with the list of keys sourced from the given Context instead of passed in.

Reply:
Good point, updated the code to reflect that. I'm trying to find all keys in the registered `ThreadLocalAccessor`s and iterate over those through the context to see if there's a value registered under that key